tests/test_list.py

```diff
@@ -2,21 +2,30 @@
 from unittest import TestCase
 
 from approvaltests import approvals
-from approvaltests.reporters.clipboard_reporter import CommandLineReporter 
-from approvaltests.reporters.multi_reporter import MultiReporter 
-from approvaltests.reporters.diff_reporter import DiffReporter 
+from approvaltests.reporters.clipboard_reporter import CommandLineReporter
+from approvaltests.reporters.multi_reporter import MultiReporter
+from approvaltests.reporters.diff_reporter import DiffReporter
 from approvaltests.reporters.generic_diff_reporter import GenericDiffReporter
 
 
 class TestList(TestCase):
     def setUp(self):
         reporter = MultiReporter(GenericDiffReporter.create('diff'), CommandLineReporter())
         approvals.set_default_reporter(reporter)
-        
+
     def test(self):
         alist = ['a', 'b', 'c', 'd', 'e']
         approvals.verify_all('letters', alist, reporter=DiffReporter())
 
     def test_uppercase(self):
         alist = ['a', 'b', 'c', 'd']
         approvals.verify_all('uppercase', alist, lambda x: '{0} => {1}'.format(x, x.upper()))
+
+    def test_format_line_part1(self):
+        alist = ['1', '2', '3']
+        approvals.verify_all('index', alist, reporter=DiffReporter())
+
+    def test_format_line_part2(self):
+        # Depends on test_format_line_part1
```

Comment:
I think it would be good to expand on this a little so that it's obvious why this test is here, e.g. 

"This is a part 2 of a test that would reproduce a bug where verify_all does not reset index when called from two different tests (see issue 32: https://github.com/approvals/ApprovalTests.Python/issues/32)".

What do you think?

As this PR fixes the bug an I have no further comments I will merge this now and this comment can be changed in a separate PR.